internal/handler/site/middleware/notfound_test.go

```diff
@@ -0,0 +1,156 @@
+package middleware_test
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"io"
+	"io/fs"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/oursky/pageship/internal/config"
+	"github.com/oursky/pageship/internal/handler/site/middleware"
+	"github.com/oursky/pageship/internal/site"
+	"github.com/stretchr/testify/assert"
+)
+
+type notFoundMockHandler struct {
+	publicFS site.FS
+}
+
+func (mh notFoundMockHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	rsc, _ := mh.publicFS.Open(r.Context(), r.URL.Path)
+	http.ServeContent(w, r, path.Base(r.URL.Path), time.Now(), rsc)
+}
+
+type RSCAdapter struct {
+	*bytes.Reader
+}
+
+func (rsca RSCAdapter) Close() error {
+	return nil
+}
+
+type FSAdapter struct {
+	embed.FS
+	subdir string
+}
+
+func (fa FSAdapter) Open(c context.Context, s string) (io.ReadSeekCloser, error) {
+	f, _ := fa.FS.Open(path.Join(fa.subdir, s))
+	b, _ := io.ReadAll(f)
+	return RSCAdapter{bytes.NewReader(b)}, nil
+}
+
+func (fa FSAdapter) Stat(s string) (*site.FileInfo, error) {
+	st, err := fs.Stat(fa.FS, path.Join(fa.subdir, s))
+	if err != nil {
+		return nil, err
+	}
+	return &site.FileInfo{
+		IsDir:       st.IsDir(),
+		ModTime:     st.ModTime(),
+		Size:        st.Size(),
+		ContentType: "",
+		Hash:        "",
+	}, err
+}
+
+//go:embed testdata/testrootwith404
+var testrootwith404FS embed.FS
```

Comment:
Please update the test cases to test that existing file would be routed unchanged.